smokefree/sites/all/modules/custom/icfi_motion_ai_api/icfi_motion_ai_api.module

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/**
+ * Implements hook_menu().
+ */
+function icfi_motion_ai_api_menu() {
+  $items = array();
+  $items['endpoint/include'] = array(
+    'page callback' => '_get_random_include_as_json',
+    'access callback' => TRUE,
+  );
+  return $items;
+}
+
+/**
+ * Callback for the API endpoint.
+ *
+ * It checks if the API key is valid, the term is valid
+ * and then returns the appropriate include data
+ */
+function _get_random_include_as_json() {
+
+  // Check if the API key is valid
+  if (is_valid_key() === FALSE) {
+    _error_response('403 Forbidden', 'Please supply a valid API key.');
+  }
+
+  $category_tid = get_category_tid();
+  // Check if the category exists
+  if ($category_tid === FALSE) {
+    _error_response('404 Not Found', 'Please supply a valid category.');
+  }
+  else {
+    $include_id = get_random_include_by_category($category_tid);
+
+    // In the unlikely event that the category does not have any include associated with it
+    if (empty($include_id)) {
+      print drupal_json_encode(array('error_message' => 'Could not find any includes for the supplied term.'));
+    }
+    else {
+      $content = _get_loaded_include($include_id, $_GET['category']);
+      print drupal_json_encode($content);
+    }
+    drupal_exit();
+  }
+}
+
+/**
+ * Check if the supplied API key is valid.
+ * Please see README.txt file of icfi_api_keys module.
+ *
+ * @return bool TRUE is the key matches the one in the database, FALSE otherwise.
+ */
+function is_valid_key() {
```

Comment:
Three of the functions start with underscores; I know this is the old way to help denote public/private functions before PHP supported it, and I use it myself, I'm just not familiar enough with the way the rest of the modules associated with this one will work to know if the underscored/not-underscored are accurate. Could you verify?

Reply:
Right. It didn't change the actual scope, but it was just used as a hint to devs. And I was just being overly picky. From what I can tell, this looks to me.